In [1]:
import pandas as pd
import numpy as np


In [2]:
df = pd.read_csv("dataset.csv")

In [3]:
df

Link  \
0     https://www.imvbox.com/watch-persian-movie-ira...   
1     https://www.imvbox.com/watch-persian-movie-ira...   
2     https://www.imvbox.com/watch-persian-movie-ira...   
3     https://www.imvbox.com/watch-persian-movie-ira...   
4     https://www.imvbox.com/watch-persian-movie-ira...   
...                                                 ...   
1432  https://www.imvbox.com/watch-persian-movie-ira...   
1433  https://www.imvbox.com/watch-persian-movie-ira...   
1434  https://www.imvbox.com/watch-persian-movie-ira...   
1435  https://www.imvbox.com/watch-persian-movie-ira...   
1436  https://www.imvbox.com/watch-persian-movie-ira...   

                      EN_title       PENGLISH_title   PERSIAN_title  \
0            Local Anaesthetic     Bi Hessie Mozeie    بی‌حسی موضعی   
1                  Disturbance            Ashoftegi        آشفته گی   
2                    Highlight              Haylayt         هایلایت   
3                        Gilda               Geelda           گیلدا   
4           Atmosphere Station     Istgahe Atmosfer  ایستگاه اتمسفر   
...                        ...                  ...             ...   
1432                   The Cow                  Gav             گاو   
1433                 Cold Iron          Sardie Ahan        سردی آهن   
1434  Secret of Behjat's House  Raze Khaneye Behjat   راز خانه بهجت   
1435              The Sicilian             Sisiliha       سیسیلی‌ها   
1436                 Charlatan            Sharlatan        شارلاتان   

                                              Content_1  \
0     جلال‌، دانشجوی سابق رشته فلسفه، متوجه می‌شود خ...   
1     «آشفته‌گی» رئالیستی و اجتماعی نیست. یک فیلم اس...   
2     یک تصادف اتومبیل آدم‌هایی را در تقابل با هم قر...   
3     گیلدا ماجرای زنی به نام «گیلدا» را روایت می کن...   
4     این فیلم روایت گر داستان زندگی زوج جوانی به اس...   
...                                                 ...   
1432  گاو "مشهدی حسن " می میرد و او که گاوش را خیلی ...   
1433                                                NaN   
1434                                                NaN   
1435  سهراب به سیاوش که کلاهبردار است و عاشق ایتالیا...   
1436  حسن و داود، دو دوست دیرینه، با هم زندگی می کنن...   

                                              Content_2  Score  Year  \
0     Jalal, a dropouts philosophy student, realizes...    4.8  2018   
1     After the murder of his rich twin brother, Bar...    3.8  2018   
2     A man and a woman are have a car accident and ...    4.4  2017   
3     Gilda who owns a restaurant has a terrible nig...    3.8  2018   
4     Vahid and Marjan are a young couple who have g...    5.6  2017   
...                                                 ...    ...   ...   
1432  Often credited as being the first Iranian New ...    8.1  1969   
1433  'Sardie Ahan' is an artistically filmed portra...    8.0  1968   
1434  After many years away, Ali Naghi comes home to...    4.7  2014   
1435  Though he comes from a poor area south of Tehr...    5.4  2011   
1436  Enjoyably absurd comedy about a young screenpl...    7.6  2008   

             Genre Time  
0            Drama   73  
1            Crime   78  
2            Drama   77  
3            Drama   79  
4            Drama   85  
...            ...  ...  
1432         Drama  100  
1433  Experimental   19  
1434         Drama   87  
1435         Crime   89  
1436        Comedy   89  

[1437 rows x 10 columns]

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1437 entries, 0 to 1436
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Link            1437 non-null   object 
 1   EN_title        1437 non-null   object 
 2   PENGLISH_title  1432 non-null   object 
 3   PERSIAN_title   1436 non-null   object 
 4   Content_1       1076 non-null   object 
 5   Content_2       1231 non-null   object 
 6   Score           1437 non-null   float64
 7   Year            1437 non-null   int64  
 8   Genre           1437 non-null   object 
 9   Time            1429 non-null   object 
dtypes: float64(1), int64(1), object(8)
memory usage: 112.4+ KB


In [5]:
df.Time

0        73
1        78
2        77
3        79
4        85
       ... 
1432    100
1433     19
1434     87
1435     89
1436     89
Name: Time, Length: 1437, dtype: object

We can see that the time cloumn has some null values and also its type is object! Where it should be int. So we have to fix this

In [6]:
df.dropna(subset=['Time'], inplace=True) #Drop null values

In [7]:
df.info() #Now we can see we have no null values but now we need to change its type to int

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1429 entries, 0 to 1436
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Link            1429 non-null   object 
 1   EN_title        1429 non-null   object 
 2   PENGLISH_title  1424 non-null   object 
 3   PERSIAN_title   1428 non-null   object 
 4   Content_1       1071 non-null   object 
 5   Content_2       1227 non-null   object 
 6   Score           1429 non-null   float64
 7   Year            1429 non-null   int64  
 8   Genre           1429 non-null   object 
 9   Time            1429 non-null   object 
dtypes: float64(1), int64(1), object(8)
memory usage: 122.8+ KB


In [8]:
df["Time"] = pd.to_numeric(df["Time"], errors='coerce')

In [9]:
df.dtypes

Link               object
EN_title           object
PENGLISH_title     object
PERSIAN_title      object
Content_1          object
Content_2          object
Score             float64
Year                int64
Genre              object
Time              float64
dtype: object

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1429 entries, 0 to 1436
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Link            1429 non-null   object 
 1   EN_title        1429 non-null   object 
 2   PENGLISH_title  1424 non-null   object 
 3   PERSIAN_title   1428 non-null   object 
 4   Content_1       1071 non-null   object 
 5   Content_2       1227 non-null   object 
 6   Score           1429 non-null   float64
 7   Year            1429 non-null   int64  
 8   Genre           1429 non-null   object 
 9   Time            1427 non-null   float64
dtypes: float64(2), int64(1), object(7)
memory usage: 122.8+ KB


In [11]:
df.dropna(subset=['Time'], inplace=True) #We drop null values again

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1427 entries, 0 to 1436
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Link            1427 non-null   object 
 1   EN_title        1427 non-null   object 
 2   PENGLISH_title  1422 non-null   object 
 3   PERSIAN_title   1426 non-null   object 
 4   Content_1       1070 non-null   object 
 5   Content_2       1226 non-null   object 
 6   Score           1427 non-null   float64
 7   Year            1427 non-null   int64  
 8   Genre           1427 non-null   object 
 9   Time            1427 non-null   float64
dtypes: float64(2), int64(1), object(7)
memory usage: 122.6+ KB


Now we have 1427 movies with EN_title, score, year, genre and time

In [14]:
df #Let's take a look at dataframe

Link  \
0     https://www.imvbox.com/watch-persian-movie-ira...   
1     https://www.imvbox.com/watch-persian-movie-ira...   
2     https://www.imvbox.com/watch-persian-movie-ira...   
3     https://www.imvbox.com/watch-persian-movie-ira...   
4     https://www.imvbox.com/watch-persian-movie-ira...   
...                                                 ...   
1432  https://www.imvbox.com/watch-persian-movie-ira...   
1433  https://www.imvbox.com/watch-persian-movie-ira...   
1434  https://www.imvbox.com/watch-persian-movie-ira...   
1435  https://www.imvbox.com/watch-persian-movie-ira...   
1436  https://www.imvbox.com/watch-persian-movie-ira...   

                      EN_title       PENGLISH_title   PERSIAN_title  \
0            Local Anaesthetic     Bi Hessie Mozeie    بی‌حسی موضعی   
1                  Disturbance            Ashoftegi        آشفته گی   
2                    Highlight              Haylayt         هایلایت   
3                        Gilda               Geelda           گیلدا   
4           Atmosphere Station     Istgahe Atmosfer  ایستگاه اتمسفر   
...                        ...                  ...             ...   
1432                   The Cow                  Gav             گاو   
1433                 Cold Iron          Sardie Ahan        سردی آهن   
1434  Secret of Behjat's House  Raze Khaneye Behjat   راز خانه بهجت   
1435              The Sicilian             Sisiliha       سیسیلی‌ها   
1436                 Charlatan            Sharlatan        شارلاتان   

                                              Content_1  \
0     جلال‌، دانشجوی سابق رشته فلسفه، متوجه می‌شود خ...   
1     «آشفته‌گی» رئالیستی و اجتماعی نیست. یک فیلم اس...   
2     یک تصادف اتومبیل آدم‌هایی را در تقابل با هم قر...   
3     گیلدا ماجرای زنی به نام «گیلدا» را روایت می کن...   
4     این فیلم روایت گر داستان زندگی زوج جوانی به اس...   
...                                                 ...   
1432  گاو "مشهدی حسن " می میرد و او که گاوش را خیلی ...   
1433                                                NaN   
1434                                                NaN   
1435  سهراب به سیاوش که کلاهبردار است و عاشق ایتالیا...   
1436  حسن و داود، دو دوست دیرینه، با هم زندگی می کنن...   

                                              Content_2  Score  Year  \
0     Jalal, a dropouts philosophy student, realizes...    4.8  2018   
1     After the murder of his rich twin brother, Bar...    3.8  2018   
2     A man and a woman are have a car accident and ...    4.4  2017   
3     Gilda who owns a restaurant has a terrible nig...    3.8  2018   
4     Vahid and Marjan are a young couple who have g...    5.6  2017   
...                                                 ...    ...   ...   
1432  Often credited as being the first Iranian New ...    8.1  1969   
1433  'Sardie Ahan' is an artistically filmed portra...    8.0  1968   
1434  After many years away, Ali Naghi comes home to...    4.7  2014   
1435  Though he comes from a poor area south of Tehr...    5.4  2011   
1436  Enjoyably absurd comedy about a young screenpl...    7.6  2008   

             Genre   Time  
0            Drama   73.0  
1            Crime   78.0  
2            Drama   77.0  
3            Drama   79.0  
4            Drama   85.0  
...            ...    ...  
1432         Drama  100.0  
1433  Experimental   19.0  
1434         Drama   87.0  
1435         Crime   89.0  
1436        Comedy   89.0  

[1427 rows x 10 columns]

In [15]:
df.describe()

Score         Year         Time
count  1427.000000  1427.000000  1427.000000
mean      6.528942  2007.177295    86.166783
std       1.562071     9.347512    19.854525
min       0.000000  1968.000000     7.000000
25%       5.400000  2001.000000    81.000000
50%       6.800000  2010.000000    90.000000
75%       8.000000  2015.000000    96.000000
max      10.000000  2020.000000   178.000000

Hear we can see the information about our numeric columns. The avarage score of a movie is around 6.5 and we can see that we have movies from year 1968 to 2020. The time column is intresting too. The avarage time of movies is about 86 minutes but the shortest movies were only 7 minutes and also the longest on is 178 minutes - near 3 hours - .

In [16]:
shortest_movies = df[df['Time'] == 7]
shortest_movies

Link    EN_title  \
659   https://www.imvbox.com/watch-persian-movie-ira...  Burnt City   
1095  https://www.imvbox.com/watch-persian-movie-ira...    The Star   

      PENGLISH_title PERSIAN_title Content_1  \
659   Shahre Sukhteh     شهر سوخته       NaN   
1095         Setareh         ستاره       NaN   

                                              Content_2  Score  Year  \
659   Soudabeh Moradian's short documentary is an ex...    7.0  2012   
1095  The musical animation 'The Star' tells the sto...    6.2  2000   

             Genre  Time  
659   Experimental   7.0  
1095         Drama   7.0